### DISCUSSION TITLE
Listen on all except wan addresses for sslocal?

### DISCUSSION THREAD

**Discussion by Klaaktu** [shadowsocks/shadowsocks-rust#599] [Q&A] [answered]
What's the best way for `sslocal` on a router to listen on `lan` and `loopback` (preferably IPv4 and v6) but not `wan`?
It seems that "local_address" expects string, so one config block for each address?

**Comment by DuckSoft** [comment, marked as answer]
I guess you can use `"locals"` array in the extended JSON config. See: https://github.com/shadowsocks/shadowsocks-rust#configuration

```json
{
  ...
  "locals": [
    {"local_address": "127.0.0.1", "local_port": 1080},
    {"local_address": "::1", "local_port": 1080}
  ],
  ...
}
```
